src/bin/main.rs

```diff
@@ -12,20 +8,6 @@ async fn main() -> anyhow::Result<()> {
 
     env_logger::init();
 
-    log::info!("Serving on http://0.0.0.0:7878");
-
-    let state = State::new().context("Failed to initialze \"State\"!")?;
-    let state = Data::new(state);
-
-    HttpServer::new(move || {
-        // `Data<State>` is being cloned to avoid creating
-        // one `State` instance per worker.
-        // Otherwise, `Data<State>` will be intialized as many
-        // times as the total count of workers configured
-        App::new().app_data(state.clone()).configure(router)
-    })
-    .bind("0.0.0.0:7878")?
-    .run()
-    .await?;
+    run("0.0.0.0:7878")?.await?;
```

Comment:
If the suggestion above is applied, then we would have more context on what this function really does, having something similar to:

```rust
server::run("0.0.0.0:7878".parse())?.await?;
```